### PR TITLE
Update elastic-agent-system-metrics dep

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This updates the elastic-agent-system-metrics to [v0.5.0](https://github.com/elastic/elastic-agent-system-metrics/releases/tag/v0.5.0) to bring in https://github.com/elastic/elastic-agent-system-metrics/pull/77 and https://github.com/elastic/elastic-agent-system-metrics/pull/73 ;

Don't merge this just yet, still want to manually build metricbeat and make sure nothing else broke, as this is a fairly large update.

## Why is it important?

This brings in a number of bug fixes.

### CODE REVIEW

**Comment by fearful-symmetry** [comment]
Update: this will have to wait on https://github.com/elastic/beats/pull/34456

File: NOTICE.txt
Comment (ycombinator) [inline NOTICE.txt:12357]: Should we release v1.9.1 of `go-sysinfo` first?

**Comment by ycombinator** [comment]
> Update: this will have to wait on https://github.com/elastic/beats/pull/34456

Hey @fearful-symmetry, I'm not sure this PR needs to wait on https://github.com/elastic/beats/pull/34456.  I think we can merge this PR first and then I'll rebase https://github.com/elastic/beats/pull/34456 on `main`.

**Comment by fearful-symmetry** [comment]
@ycombinator sorry, should have been a little more specific. This update brings in the breaking changes related to FQDN support, so right now this PR won't compile, and those breaking changes are fixed in https://github.com/elastic/beats/pull/34456, and I figured it would be better to wait until that was merged rather than fix it myself, so I didn't make you or Anderson have to deal with merge conflicts.

**Comment by ycombinator** [comment]
> This update brings in the breaking changes related to FQDN support, so right now this PR won't compile, and those breaking changes are fixed in https://github.com/elastic/beats/pull/34456,

Thanks for clarifying, @fearful-symmetry.  If there's any urgency to merging this PR, please let me know and I'll extract out the necessary fixes from #34456 to unblock you.  With EAH it might be another week or so before #34456 is merged.

**Review by leehinman — APPROVED** [review]
LGTM

**Pull request merged**
